relay-server/src/envelope.rs

```diff
@@ -617,9 +617,12 @@ impl Item {
             ItemType::ReplayEvent | ItemType::ReplayRecording => Some(DataCategory::Replay),
             ItemType::ClientReport => None,
             ItemType::CheckIn => Some(DataCategory::Monitor),
+            ItemType::Span | ItemType::OtelSpan => Some(if indexed {
```

Comment:
For standalone spans, `indexed` will always be `false`, because it gets its value from https://github.com/getsentry/relay/blob/518148061bde526b7f6713b8e3c32c4660e4d53d/relay-server/src/utils/managed_envelope.rs#L299, which is only true if there was an event!

Can we live with that for now? I can try to make `indexed` depend on an item header instead of the event flag, then everything should fall into place.

Reply:
Then we should reverse the condition, have default indexed (so that standalone spans, not being dynamically sampled, without an event, are always counted as indexed) and that will do the trick until we set this field with an item header.